### PR TITLE
INTMDB-611: Update search index to get api keys from AWS secret manager

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

This PR updates search index to get api keys from AWS secret manager.
Link to any related issue(s): 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

Tested locally by running `cfn test`
```
================================================================================= 8 passed, 4 skipped, 12 deselected, 1 warning in 1170.97s (0:19:30)
```

### CODE REVIEW

File: cfn-resources/search-index/cmd/resource/resource.go [hunk 1]
```diff
@@ -34,12 +34,10 @@ func setup() {
 	util.SetupLogger("search-index")
 }
 
-var CreateRequiredFields = []string{constants.GroupID, constants.ClusterName, constants.PubKey, constants.PvtKey}
-var ReadRequiredFields = []string{constants.GroupID, constants.ClusterName, constants.IndexID, constants.PubKey, constants.PvtKey}
-var UpdateRequiredFields = []string{constants.GroupID, constants.ClusterName, constants.IndexID, constants.PubKey, constants.PvtKey}
-var DeleteRequiredFields = []string{constants.GroupID, constants.ClusterName, constants.IndexID, constants.PubKey, constants.PvtKey}
-var ListRequiredFields = []string{constants.ProjectID, constants.ClusterName, constants.Database, constants.CollectionName,
-	constants.PubKey, constants.PvtKey}
+var CreateRequiredFields = []string{constants.ProjectID, constants.ClusterName}
```
Comment (colm-quinn) [inline cfn-resources/search-index/cmd/resource/resource.go:37]: General note for once we GA, changing the parameter names would be a breaking change.

File: cfn-resources/search-index/mongodb-atlas-searchindex.json [hunk 2]
```diff
@@ -77,37 +77,35 @@
...
-      "permissions": []
+      "permissions": [
+        "secretsmanager:GetSecretValue"
+      ]
     }
   },
   "primaryIdentifier": [
-    "/properties/IndexId"
+    "/properties/IndexId",
+    "/properties/ProjectId",
```
Comment (colm-quinn) [inline cfn-resources/search-index/mongodb-atlas-searchindex.json:107]: [q] GroupId wasn't required before - was that a bug?

**Review by colm-quinn — APPROVED** [review]
LGTM, some small questions, nothing blocking.

File: cfn-resources/search-index/mongodb-atlas-searchindex.json (hunk 2 above)
Comment (andreaangiolillo) [reply cfn-resources/search-index/mongodb-atlas-searchindex.json]: good catch. Actually, the projectId is not needed as IndexId is already unique

Comment (andreaangiolillo) [reply cfn-resources/search-index/mongodb-atlas-searchindex.json]: updated

**Comment by andreaangiolillo** [comment]
I released my changes in the private repo and tested with atlas prod

Screenshot 2023-02-22 at 11 23 59
Screenshot 2023-02-22 at 11 21 57
Screenshot 2023-02-22 at 11 26 11
Screenshot 2023-02-22 at 11 26 23

**Pull request merged**